### PR TITLE
[Feature] Block Tunes API

### PR DESCRIPTION
**Pull request by gohabereg**
Things left:

- [ ] Refactor
- [ ] Write docs
- [ ] Add common tunes support
- [ ] Render common and Block tunes in separate containers

### CODE REVIEW

File: example/tunes/favorite/index.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+class FavoriteTune {
```
Comment (gohabereg) [inline example/tunes/favorite/index.js]: Temp file, will remove it after all is tested

File: example/example.html [hunk 2]
```diff
@@ -93,7 +94,8 @@
           config: {
             placeholder: 'Header'
           },
-          shortcut: 'CMD+SHIFT+H'
+          shortcut: 'CMD+SHIFT+H',
+          tunes: [FavoriteTune],
```
Comment (neSpecc) [inline example/example.html:98]: can support two ways:
```js
[
  {
    class: FavoriteTune,
    name: 'favourite'
  },
  SpoilerTune
]
```

File: src/components/block/index.ts [hunk 3]
```diff
@@ -633,7 +654,15 @@ export default class Block {
         contentNode = $.make('div', Block.CSS.content),
         pluginsContent = this.tool.render();
 
-    contentNode.appendChild(pluginsContent);
+    const wrapped = this.tunes.reduce((res, tune) => {
```
Comment (neSpecc) [inline src/components/block/index.ts:657]: need to move to a separated method and leave a comment

File: src/components/modules/api/blocks.ts [hunk 4]
```diff
@@ -174,14 +174,12 @@ export default class BlocksAPI extends Module {
...
    * @param {BlockToolData} data — Tool data to insert
-   * @param {ToolConfig} config — Tool config
    * @param {number?} index — index where to insert new Block
    * @param {boolean?} needToFocus - flag to focus inserted Block
    */
   public insert = (
     type: string = this.config.initialBlock,
     data: BlockToolData = {},
-    config: ToolConfig = {},
     index?: number,
```
Comment (neSpecc) [inline src/components/modules/api/blocks.ts:183]: is this a breaking change?

File: src/components/modules/sanitizer.ts [hunk 5]
```diff
@@ -68,9 +68,20 @@ export default class Sanitizer extends Module {
                            *//**
                               *
                               *//**
- *
- */
-(
+                                 *
```
Comment (neSpecc) [inline src/components/modules/sanitizer.ts:71]: there is a bug in jsdoc plugin with autofixing

**Comment by hotrush** [comment]
Hope anyone will look into this pr :)

**Comment by christoph-kluge** [comment]
@gohabereg can I support you somehow? I was looking into this feature yesterday and decided to hardcod my custom tune now. Would love to this see this ready soon.

**Comment by kaaaaaaaaaaai** [comment]
@gohabereg @neSpecc @hotrush 
What's going on now?
I would like this feature very much!

**Comment by neSpecc** [comment]
> @gohabereg @neSpecc @hotrush
> What's going on now?
> I would like this feature very much!

We will start working on it this month.

**Comment by hotrush** [comment]
@neSpecc sounds good 👍  waiting for this feature for about a year

**Comment by gohabereg** [comment]
New PR with this feature #1596

**Pull request closed without merging**